odoo/fields.py

```diff
@@ -1402,6 +1423,40 @@ def determine_domain(self, records, operator, value):
         """ Return a domain representing a condition on ``self``. """
         return determine(self.search, records, operator, value)
 
+    # cache operations
+    def _cache(self, env):
```

Comment:
What about `get_cache_mapping` instead?  Because `field._cache(env)` is quite obscure...
And by the way, you can make the method 100% safe by returning a read-only proxy to the cache mapping `m` as `m.keys().mapping`.